Sources/MongoSwift/MongoError.swift

```diff
@@ -179,69 +231,64 @@ internal func parseMongocError(_ error: bson_error_t, errorLabels: [String]? = n
     }
 }
 
-/// Internal function used to get an appropriate error from a server reply to a command.
-internal func getErrorFromReply(
-        bsonError: bson_error_t,
-        from reply: Document,
-        forBulkWrite bulkWrite: BulkWriteOperation? = nil,
-        withResult result: BulkWriteResult? = nil) -> MongoError {
-    // if writeErrors or writeConcernErrors aren't present, then this is likely a commandError.
-    guard reply["writeErrors"] != nil || reply["writeConcernErrors"] != nil else {
-        return parseMongocError(bsonError, errorLabels: reply["errorLabels"] as? [String])
+/// Internal function used to get an appropriate error from a libmongoc error and/or a server reply to a command.
+internal func getMongoError(error bsonError: bson_error_t,
```

Comment:
I agree that it is a bit fishy. The original reason for doing it like this is that sometimes a non-bulk write error will occur during a bulk write. If we introduced a separate `extractBulkError`, we'd have it returning things that weren't `ServerError.bulkWriteErrors` (e.g. `RuntimeError.connectionError` or `ServerError.commandError`). I also wanted to funnel all error handling through a single function so that we wouldn't need to think about which one to call to get an appropriate error.

We could solve this problem by naming the new bulk function something like `extractBulkOperationMongoError`, if you think that would be preferable. That way it wouldn't signal as much that a bulk write error is expected. Alternatively, we could overload the original function into something like `extractMongoError(for bulk: BulkWriteOperation, error: bson_error_t, reply: Document? = nil, result: BulkWriteResult? = nil)`, and leave the bulk op stuff out of the vanilla one. What do you think?

Reply:
I'm into the overloading idea. I just think the two paths are sufficiently divergent that it will improve readability to split it up.

Reply:
Refactored to `extractMongoError(bulkOp:error:reply:partialResult)`. I also moved the `getWriteError` and `getBulkWriteError` functions into the bodies of the separate code paths since they were being exclusively called there.